cosmoz-default-tree.html

```diff
@@ -23,134 +34,192 @@
-			return;
+	/**
+	 * Returns the node of a given path.
+	 * @return {object}
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {object} nodeObj (The object the search should be based on.) Default: this._treeData
+	 * @param {string} separatorSign (The string which separates the path. e.g ".") Default: "."
+	 * @param {string} childProperty (The string which describes the child property of a node.) Default: "children"
+	*/
+	Cosmoz.DefaultTree.prototype.getNodeByPathLocator = function (pathLocator, nodeObj, separatorSign, childProperty) {
+		if (!pathLocator) {
+			return this._treeData;
 		}
 
-		if (propertyName === 'pathLocator') {
-			return this._getPathByPathLocator(propertyValue);
-		}
+		// Defaults
+		nodeObj = nodeObj ? nodeObj : this._treeData;
+		separatorSign = separatorSign ? separatorSign : '.';
+		childProperty = childProperty ? childProperty : 'children';
 
-		var node = this._searchAllRoots(propertyName, propertyValue);
-		if (node !== undefined) {
-			return this._getPathByPathLocator(node.pathLocator);
-		}
+		var	pathNodes = this.getPathNodes(pathLocator, nodeObj, separatorSign, childProperty),
+			lastNodeOnPath = pathNodes.pop();
+		return lastNodeOnPath !== undefined ? lastNodeOnPath : null;
 	};
 
-	Cosmoz.DefaultTree.prototype.getChildren = function (node) {
-		if (node === undefined) {
-			return;
+	/**
+	 * Returns the nodes on a given path.
+	 * @return {array} or null if there was an empty response.
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {object} nodeObj (The object the search should be based on.) Default: this._treeData
+	 * @param {string} separatorSign (The string which separates the path. e.g ".") Default: "."
+	 * @param {string} childProperty (The string which describes the child property of a node.) Default: "children"
+	*/
+	Cosmoz.DefaultTree.prototype.getPathNodes = function (pathLocator, nodeObj, separatorSign, childProperty) {
+		if (!pathLocator) {
+			return this._roots;
 		}
-		return _objectValues(node.children);
-	};
 
-	Cosmoz.DefaultTree.prototype.getProperty = function (node, propertyName) {
-		if (node === undefined || propertyName === undefined) {
-			return;
-		}
+		// Defaults
+		separatorSign = separatorSign ? separatorSign : '.';
+		childProperty = childProperty ? childProperty : 'children';
+
+		var path = pathLocator.split(separatorSign),
+			pathSegment = nodeObj ? nodeObj : this._treeData,
+			pathIsInvalid,
+			nodes;
+		
+		// Get the nodes on the path.
+		nodes = path.map(function (nodeKey) {
+			var node = pathSegment[nodeKey],
+				children = node[childProperty];
+			if (node && children !== undefined && Object.keys(children).length > 0) {
+				pathSegment = children;
+			}
+			return node;
+		}, this);
 
-		return node[propertyName];
-	};
+		// Check if all nodes on the path are valid.
+		pathIsInvalid = nodes.some(function (n) {
```

Comment:
I haven't found it to be invalid in the current implementation.
Though, If someone would use this element with their "own" implementation of a navigator or similar it should return an "expected" (not sure if expected is the right word here) behavior.

But you are right, it actually might should be implemented in `getNodeByPathLocator` rather than in `getPathNodes`.

Imagine those nodes: 1 - 9 - 5 - 7
```
getPathNodes(1.9.33.24) //would return [1,9, undefined, undefined] => which could be ok

getNodeByPathLocator(1.9.33.24) //would return 9 in the current implementation => which would not be acceptable
```

I'll move the check to `getNodeByPathLocator`.